server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -493,7 +493,7 @@ private ClusterHealthResponse clusterHealth(
         String[] concreteIndices;
         if (request.level().equals(ClusterHealthRequest.Level.AWARENESS_ATTRIBUTES)) {
             String awarenessAttribute = request.getAwarenessAttribute();
-            concreteIndices = clusterState.getMetadata().getConcreteAllIndices();
+            concreteIndices = clusterState.getMetadata().getConcreteAllIndices().toArray(String[]::new);
```

Comment:
Does it make sense to change `concreteIndices` to lists as well? Instead of doing a back and forth to arrays?

Reply:
Addressed it in new PR https://github.com/opensearch-project/OpenSearch/pull/14723 (This PR has screwed-up after multiple rebase)